File: Escola/Cursos.Repository/Services/AvaliacaoobservacaoService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  AvaliacaoobservacaoService
      /// </summary>
      public partial class AvaliacaoobservacaoService : BaseServices, IAvaliacaoobservacaoService
      {

		public AvaliacaoobservacaoService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_AVALIACAOOBSERVACAO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_AVALIACAOOBSERVACAO> entity)
        {
            UnitOfWork.AvaliacaoobservacaoRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_AVALIACAOOBSERVACAO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_AVALIACAOOBSERVACAO entity)
        {	
            UnitOfWork.AvaliacaoobservacaoRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_AVALIACAOOBSERVACAO
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_AVALIACAOOBSERVACAO entity)
        {
            UnitOfWork.AvaliacaoobservacaoRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_AVALIACAOOBSERVACAO
        /// </summary>
        /// <returns>IEnumerable EV_AVALIACAOOBSERVACAO</returns>
        public IEnumerable<EV_AVALIACAOOBSERVACAO> GetAll()
        {
			return UnitOfWork.AvaliacaoobservacaoRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_AVALIACAOOBSERVACAO
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_AVALIACAOOBSERVACAO entity)
        {
            UnitOfWork.AvaliacaoobservacaoRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_AVALIACAOOBSERVACAO By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_AVALIACAOOBSERVACAO</returns>
		public EV_AVALIACAOOBSERVACAO GetById(int id)
        {
			return UnitOfWork.AvaliacaoobservacaoRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_AVALIACAOOBSERVACAO> FindBy(Expression<Func<EV_AVALIACAOOBSERVACAO, bool>> predicate)
        {
            return UnitOfWork.AvaliacaoobservacaoRepository.FindBy(predicate);
        }
      }
}

File: Escola/Cursos.Site/app/services/AvaliacaoobservacaoService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 5:17:26 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("AvaliacaoobservacaoService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Avaliacaoobservacao/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Avaliacaoobservacao/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Avaliacaoobservacao, callback) {
            $http.post(serviceBase + "Avaliacaoobservacao/post", Avaliacaoobservacao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Avaliacaoobservacao, callback) {
            $http.post(serviceBase + "Avaliacaoobservacao/put", Avaliacaoobservacao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Avaliacaoobservacao, callback) {
            $http.post(serviceBase + "Avaliacaoobservacao/delete", Avaliacaoobservacao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

